### PR TITLE
Update pudl v2024.5.0

### PR DESCRIPTION
**Pull request by TrentonBush**
This PR does four things:
1. Update PUDL data to v2024.5.0. This was motivated by the addition of balancing authority codes to the EIA 860m changelog.
2. Include the new BA codes in the data_mart tables `projects_current_860m` and `county_concrete_mw`. Call them `iso_region` to match the nomenclature of existing LBNL/GridStatus ISO data
3. Bugfix: split the (accidental) 860m single wind category back into separate onshore/offshore types.
4. Convert `src/dbcp/data_mart/county_concrete_mw.py:_get_concrete_aggs` from SQL to pandas, as suggested in [this comment](https://github.com/deployment-gap-model-education-fund/deployment-gap-model/pull/352#discussion_r1701525335) of PR #352

### CODE REVIEW

File: docs/data-mart/county_concrete_mw.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ This table contains county-level aggregates of in-progress power generation proj
 |Subject|Column|Description|Source|Notes|
 |----|----|----|----|----|
 |Identifiers|`resource_clean`|Fuel or project type|EIA 860m or ACP||
-||`iso_region`|The ISO/RTO the project will belong to. Usually 1:1 with county, but border counties may have multiple ISO/RTOs|EIA 860m or ACP||
+||`iso_region`|The ISO/RTO or Balancing Authority the project will belong to. Usually 1 per county, but border counties may have multiple.|EIA 860m or ACP||
```
Comment (bendnorman) [inline docs/data-mart/county_concrete_mw.md:11]: balance authority borders?

File: src/dbcp/data_mart/county_concrete_mw.py [hunk 2]
```diff
@@ -4,94 +4,105 @@
 import pandas as pd
 import sqlalchemy as sa
 
+from dbcp.data_mart.projects import get_eia860m_current
 from dbcp.helpers import get_sql_engine
 
 
 def _get_concrete_aggs(engine: sa.engine.Engine) -> pd.DataFrame:
```
Comment (bendnorman) [inline src/dbcp/data_mart/county_concrete_mw.py:11]: Yay thanks for migrating this to pandas!

File: src/dbcp/validation/tests.py [hunk 3]
```diff
@@ -243,9 +243,10 @@ def test_county_wide_coverage(engine: Engine):
         df.shape[0] == n_counties
     ), "counties_wide_format does not contain all counties"
     notnull = df.notnull()
-    assert (
-        notnull.any(axis=1).sum() == 2457
-    ), f"counties_wide_format has unexpected county coverage: {notnull[notnull.any(axis=1)]}"
+    assert notnull.any(axis=1).sum() == 2461, (
```
Comment (bendnorman) [inline src/dbcp/validation/tests.py:246]: Any idea why this went up?

**Review by bendnorman — APPROVED** [review]

File: src/dbcp/data_mart/county_concrete_mw.py (hunk 2 above)
Comment (bendnorman) [inline src/dbcp/data_mart/county_concrete_mw.py:11]: A doc string would be helpful here.

File: src/dbcp/validation/tests.py (hunk 3 above)
Comment (TrentonBush) [reply src/dbcp/validation/tests.py:246]: The PUDL update included about 300 more rows in `core_eia__entity_plants` for some reason. Apparently some of those were in new counties, or the county assignments changed. I didn't make any code changes to that pipeline

File: docs/data-mart/county_concrete_mw.md (hunk 1 above)
Comment (TrentonBush) [reply docs/data-mart/county_concrete_mw.md:11]: Ya, if there are two sets of wires in the same county, both BAs will show up. Here's [GridStatus's map](https://www.gridstatus.io/ba-map) (I think based on EIA BA:county data)

**Pull request merged**